File: php/testcode/weakhash_016.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakhash_argon2id
function weakhash016(BenchmarkRequest $req): BenchmarkResponse {
    $password = $req->post('password');
    $hash = password_hash($password, PASSWORD_ARGON2ID); // vuln-code-snippet safe-line php_weakhash_argon2id
    return BenchmarkResponse::json(['hash' => $hash]);
}
// vuln-code-snippet end php_weakhash_argon2id

File: ruby/testcode/weakhash_016.rb
require_relative 'shared'

module SCrypt
  class Password
    def self.create(password, opts = {}); "scrypt:#{password}"; end
  end
end

# vuln-code-snippet start ruby_weakhash_scrypt_create
def hash_password_scrypt(req)
  password = req.param('password')
  hashed = SCrypt::Password.create(password) # vuln-code-snippet safe-line ruby_weakhash_scrypt_create
  BenchmarkResponse.ok(hashed)
end
# vuln-code-snippet end ruby_weakhash_scrypt_create
